src/types.rs

```diff
@@ -73,36 +75,40 @@ impl<'de> Deserialize<'de> for Proof {
 
 impl core::fmt::Debug for Proof {
     fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-        // TODO: implement
-        Ok(())
+        self.0[..].fmt(f)
```

Comment:
We should agree on a format for the binary. I believe this would dump `[0, 0, …]` where the other custom `Debug` I saw try to dump hex string.

What would be more useful to us?

Reply:
I think the hex version would be easier to read - I'll conform it to the other implementations